Squirrel/Controllers/AccountController.cs

```diff
@@ -1,17 +1,165 @@
+
+            if (await roleManager.FindByNameAsync("service") == null)
+                await roleManager.CreateAsync(new IdentityRole("service"));
+
+            if (await userManager.FindByEmailAsync(adminEmail) == null)
+            {
+                User admin = new() { Email = adminEmail, UserName = adminEmail};
+                IdentityResult result = await userManager.CreateAsync(admin, password);
+                if (result.Succeeded)
+                {
+                    await userManager.ConfirmEmailAsync(admin!, await userManager.GenerateEmailConfirmationTokenAsync(admin));
+                    await userManager.AddToRoleAsync(admin, "admin");
+                }
+            }
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> AuthenticateAsync(LoginModel model)
+        {
+            var result = await _signInManager.PasswordSignInAsync(model.Email, model.Password, true, false);
+            return result.Succeeded ? Ok() : BadRequest(new[] { _localizer["Invalid login and/or password"].Value });
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> RegisterAsync(RegisterModel model, string callbackUrl)
+        {
+            if (model.Same)
+            {
+                User user = new(model);
+                var result = await _userManager.CreateAsync(user, model.Password);
+                if (result.Succeeded)
+                {
+                    var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
+                    callbackUrl = Url.ActionLink("ConfirmEmail", "Account", new { user.Id, code, callbackUrl });
+                    await _emailSender.SendEmailAsync(email: model.Email,
+                                                     subject: _localizer["Confirm your email"].Value,
+                                                     htmlMessage: _localizer["Please confirm your account by <a href='{0}'>clicking here</a>.", HtmlEncoder.Default.Encode(callbackUrl!)].Value);
+
+                    return Ok();
+                }
+                else
+                    return BadRequest(result.Errors.Select(e => e.Description));
+            }
+            else
+                return BadRequest(new[] { _localizer["Password are not the same"].Value });
+        }
+
+        [Authorize]
+        [HttpPost]
+        public async Task<IActionResult> LogoutAsync()
+        {
+            await _signInManager.SignOutAsync();
+            return Ok();
+        }
+
+        [Authorize]
+        [HttpGet]
+        public async Task<IActionResult> EmailAsync() => Ok((await _userManager.GetUserAsync(User)).Email);
+
+        [Authorize]
+        [HttpGet]
+        public async Task<IActionResult> CurrencyAsync() => Ok((await _userManager.GetUserAsync(User)).Currency);
+
+
+        [HttpPatch]
+        public async Task<IActionResult> ConfirmEmailAsync(string Id, string code, string callbackUrl)
```

Comment:
Does such endpoints, which contain several parameters will work with json objects?
Or where these parameters are expected to be passed from? Thanks for your answer

Reply:
They are expected to be passed from request URL, like
/api/Account/ConfirmEmail?id=id&code=code&callbackUrl=url